test/plugins/features/movies/controller.test.js

```diff
@@ -1,19 +1,37 @@
 'use strict';
 
-const Controller   = require('../../../../lib/plugins/features/movies/controller');
-const Knex         = require('../../../../lib/libraries/knex');
-const Movie        = require('../../../../lib/models/movie');
-const MovieFactory = require('../../../factories/movie');
+const Bluebird = require('bluebird');
+
+const Controller      = require('../../../../lib/plugins/features/movies/controller');
+const Knex            = require('../../../../lib/libraries/knex');
+const LocationFactory = require('../../../factories/location');
+const Movie           = require('../../../../lib/models/movie');
+const MovieFactory    = require('../../../factories/movie');
 
 const movie1 = MovieFactory.build({ name: 'Star Trek', release_year: 1982 });
 const movie2 = MovieFactory.build({ name: 'Star Trek 2', release_year: 1984 });
 const movie3 = MovieFactory.build({ name: 'Star Trok', release_year: 1986 });
 
+const location1 = LocationFactory.build({ name: 'Hayes Valley' });
+
 describe('movie controller', () => {
 
   beforeEach(() => {
-    return Knex.raw('TRUNCATE movies CASCADE')
-    .then(() => Knex('movies').insert([movie1, movie2, movie3]));
+
+    return Bluebird.all([
+      Knex.raw('TRUNCATE movies CASCADE'),
+      Knex.raw('ALTER SEQUENCE movies_id_seq RESTART'),
+      Knex.raw('TRUNCATE locations CASCADE'),
+      Knex.raw('ALTER SEQUENCE locations_id_seq RESTART'),
+      Knex.raw('TRUNCATE locations_movies CASCADE')
+    ])
+    .then(() => {
+      Bluebird.all([
```

Comment:
You need to make sure you `return` the promise here so that it'll be attached to the primary promise chain. If it's not attached to the promise chain, it won't be waited on before moving on, which means the test could potentially execute before the insert finishes. This would cause your tests to fail non-deterministically.